CHANGELOG.md

```diff
@@ -1,5 +1,337 @@
+  [pull/333]: https://github.com/cloudflare/wrangler/pull/333
+  [kv]: https://www.cloudflare.com/products/workers-kv/
+
+  - **Always use multipart upload - [ashleymichal], [issue/280][pull/328]**
+
+  We've updated how Wrangler uploads projects to the Workers runtime - as Wrangler supports the entire Workers ecosystem (including tools like [Workers KV][kv], the publishing process should use [multipart forms](https://dev.to/sidthesloth92/understanding-html-form-encoding-url-encoded-and-multipart-forms-3lpa) to allow uploading all of the data that represents a Workers project in a single step.
+
+  [ashleymichal]: https://github.com/ashleymichal
+  [issue/280]: https://github.com/cloudflare/wrangler/issue/280
+  [pull/328]: https://github.com/cloudflare/wrangler/pull/328
+
+  - **Better error printing - [granjef3], [pull/327]**
+
+  We've updated how we log errors in Wrangler's output to make it a little easier to read. If you're a Rust developer interested in _how_ we did this, check out the [pull request][pull/327]!
+
+  [granjef3]: https://github.com/granjef3
+  [pull/327]: https://github.com/cloudflare/wrangler/pull/327
+
+  - **Add message module for terminal output - [ashleymical], [issue/219][pull/263]**
+
+    We've made improvements to Wrangler's terminal output functionality, with support for various log levels and implementations in Wrangler's API for easily using the log levels in future development.
+
+    The new terminal output functionality can be used by importing the `terminal::message` crate:
+
+    ```rust
+    use crate::terminal::message;
+
+    message::info("Building project") // "💁‍ Building project"
+    message::success("Your project has been deployed!") // "✨ Your project has been deployed!"
+
+    // Other available functions:
+    // message::warn, message::user_error, message::service_error, message::working, message::preview
+    ```
+
+    [ashleymical]: https://github.com/ashleymichal
+    [issue/219]: https://github.com/cloudflare/wrangler/issues/219
+    [pull/263]: https://github.com/cloudflare/wrangler/pull/263
+
+  - **Adds more descriptive subdomain errors - [EverlastingBugstopper], [issue/207][pull/259]**
+
+    It's super easy to grab a workers.dev subdomain using the `subdomain` command in `wrangler` – so easy, in fact, that many people were trying to use it without even having a Cloudflare account! `wrangler` now warns users when they attempt to add a subdomain without configuring their `account_id` in `wrangler.toml`, as well as when you've already registered a subdomain, or if the subdomain you're trying to register has already been claimed.
+
+    [EverlastingBugstopper]: https://github.com/EverlastingBugstopper
+    [issue/207]: https://github.com/cloudflare/wrangler/issue/207
+    [pull/259]: https://github.com/cloudflare/wrangler/pull/259
+
+  - **Allow custom webpack configuration in wrangler.toml - [EverlastingBugstopper], [issue/246][pull/253]**
```

Comment:
feature

Reply:
(copied from other one) ah, this is in the feature section! is that right or did you want it somewhere else?

Reply:
apparently another nonsense comment. i should read in the rendered view.. i think i'm getting lost